examples/helm/ceph/templates/mon/statefulset.yaml

```diff
@@ -0,0 +1,119 @@
+---
+apiVersion: apps/v1beta1
+kind: StatefulSet
+metadata:
+  labels:
+    version: v0.1.0
+    app: ceph
+    daemon: mon
+    release: {{ .Release.Name }}
+  name: ceph-mon
+spec:
+  serviceName: {{ .Values.service.mon.name | quote }}
+  replicas: {{ .Values.resources.mon.replicas }}
+  template:
+    metadata:
+      name: ceph-mon
+      labels:
+        version: v0.1.0
+        app: ceph
+        daemon: mon
+        release: {{ .Release.Name }}
+      annotations:
+        scheduler.alpha.kubernetes.io/affinity: >
+          {
+            "podAntiAffinity": {
+              "RequiredDuringSchedulingIgnoredDuringExecution": [{
+                "labelSelector": {
+                  "matchExpressions": [{
+                    "key": "daemon",
+                    "operator": "In",
+                    "values":["mon"]
+                  }]
+                },
+              "topologyKey": "kubernetes.io/hostname"
+              }]
+            }
+          }
+    spec:
+      serviceAccount: default
+      containers:
+        - name: main
+          image: {{ .Values.images.daemon }}
+          imagePullPolicy: {{ .Values.image_policy.pull }}
+          lifecycle:
+            preStop:
+                exec:
+                  command:
+                    - /remove-mon.sh
+          ports:
+            - containerPort: {{ .Values.service.mon.port }}
+          env:
+            - name: CEPH_DAEMON
+              value: MON
+            - name: KV_TYPE
+              value: k8s
+            - name: NETWORK_AUTO_DETECT
+              value: "1"
```

Comment:
@hunter This is an excellent suggestion. @leseb using the method Hunter has would be a nice improvement I think.
``` yaml
            - name: NETWORK_AUTO_DETECT
              value: "0"
            - name: MON_IP
              valueFrom:
                fieldRef:
                  fieldPath: status.podIP
```